nixos/modules/services/mail/mailpit.nix

```diff
@@ -0,0 +1,106 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (config.services.mailpit) instances;
+  inherit (lib)
+    cli
+    concatStringsSep
+    const
+    filterAttrs
+    getExe
+    mapAttrs'
+    mkIf
+    mkOption
+    nameValuePair
+    types
+    ;
+
+  isNonNull = v: v != null;
+  genCliFlags =
+    settings: concatStringsSep " " (cli.toGNUCommandLine { } (filterAttrs (const isNonNull) settings));
+in
+{
+  options.services.mailpit.instances = mkOption {
+    default = { };
+    type = types.attrsOf (
+      types.submodule {
+        freeformType = types.attrsOf (
+          types.oneOf [
+            types.str
+            types.int
+            types.bool
+          ]
+        );
+        options = {
+          database = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            example = "mailpit.db";
+            description = ''
+              Specify the local database filename to store persistent data.
+              If `null`, a temporary file will be created that will be removed when the application stops.
+              It's recommended to specify a relative path. The database will be written into the service's
+              state directory then.
+            '';
+          };
+          max = mkOption {
+            type = types.ints.unsigned;
+            default = 500;
+            description = ''
+              Maximum number of emails to keep. If the number is exceeded, old emails
+              will be deleted.
+
+              Set to `0` to never prune old emails.
+            '';
+          };
+          listen = mkOption {
+            default = "0.0.0.0:8025";
+            type = types.str;
+            description = ''
+              HTTP bind interface and port for UI.
+            '';
+          };
+          smtp = mkOption {
+            default = "0.0.0.0:1025";
```

Comment:
Maybe default to `127.0.0.1:1025`. See https://github.com/NixOS/nixpkgs/pull/329657#issuecomment-2288699086